### PR TITLE
Removed typings.json, added @types/* devDependencies and made tsconfig.json friendlier.

### PR DESCRIPTION
**Pull request by usergenic**
Have tested npm run builld after npm installing for both node 4 and 6 and this appears to build happily now.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,33 +3,24 @@
...
-        "src/util.ts",
-        "typings/main.d.ts"
+    "include": [
+        "src/**/*.ts",
+        "custom_typings/*.d.ts"
     ],
-    "atom": {
-        "rewriteTsconfig": true
-    }
+    "exclude": [
```
Comment (justinfagnani) [inline tsconfig.json]: shouldn't be necessary

File: package.json [hunk 2]
```diff
@@ -42,8 +42,10 @@
     "semver": "^5.1.0"
   },
   "devDependencies": {
-    "typescript": "^1.9.0-dev.20160418",
-    "typings": "^0.6.8",
+    "@types/progress": "^1.1.8",
```
Comment (justinfagnani) [inline package.json]: don't need node or yaml?

Comment (usergenic) [inline package.json]: No build errors so I assume those were unnecessary.

Comment (usergenic) [inline package.json]: Also I should mention I did actually `node run lib/tattoo.js` after build to verify this works.

Runs fine in node 4.  I have build issues with nodegit and node 6 due to openssl or something so haven't verified node 6 is happy yet.

File: tsconfig.json (hunk 1 above)
Comment (usergenic) [inline tsconfig.json]: Oh yeah that's silly.  Removed.

**Pull request merged**